docs/src/man/basics.md

```diff
@@ -0,0 +1,2042 @@
+
+julia> data.marks === x # no copy is performed
+true
+
+julia> data[1:3, :semester] = [2, 4, 6] # set value of `:semester` in row `1:3` to values `[2, 4, 6]` *in-place*
+3-element Vector{Int64}:
+ 2
+ 4
+ 6
+
+julia> data
+6×3 DataFrame
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   1 │    80  English                  2
+   2 │    85  Computer Science         4
+   3 │    98  Mathematics              6
+   4 │    95  English                  4
+   5 │    78  Computer Science         5
+   6 │    89  Mathematics              6
+
+julia> data[!, :subject] = ["English", "Computer Science", "Mathematics", "History", "Economics", "Psychology"] # replaces `:subject` without copying
+6-element Vector{String}:
+ "English"
+ "Computer Science"
+ "Mathematics"
+ "History"
+ "Economics"
+ "Psychology"
+
+julia> data
+6×3 DataFrame
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   1 │    80  English                  2
+   2 │    85  Computer Science         4
+   3 │    98  Mathematics              6
+   4 │    95  History                  4
+   5 │    78  Economics                5
+   6 │    89  Psychology               6
+
+julia> data[3, 1:3] = [78, "Economics", 7] # set row `3` of cols `1:3(:marks, :subject, :semester)` in-place
+3-element Vector{Any}:
+ 78
+   "Economics"
+  7
+
+julia> data
+6×3 DataFrame
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   1 │    80  English                  2
+   2 │    85  Computer Science         4
+   3 │    78  Economics                7
+   4 │    95  History                  4
+   5 │    78  Economics                5
+   6 │    89  Psychology               6
+
+julia> dfr = data[2, :] # passed `:`(a column) in a parent DataFrame to get the *DataFrameRow*
+DataFrameRow
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   2 │    85  Computer Science         4
+
+julia> dfr
+DataFrameRow
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   2 │    85  Computer Science         4
+
+julia> dfr.marks = 98 # set value of col `:marks` in row `2` to `98` in-place
+98
+
+julia> dfr
+DataFrameRow
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   2 │    98  Computer Science         4
+
+julia> dfr[2:3] = ["Mathematics", 5] # set values of entries in columns `:subject` and `:semester` 
+2-element Vector{Any}:
+  "Mathematics"
+ 5
+
+julia> dfr
+DataFrameRow
+ Row │ marks  subject      semester
+     │ Int64  String       Int64
+─────┼──────────────────────────────
+   2 │    98  Mathematics         5
+
+julia> sdf = view(data, :, 2:3) # Column subsetting
```

Comment:
you should explain views before making this example

Reply:
Yeah, here I have taken views because I was discussing `setindex!` and I wanted to talk about SubDataFrame as well. I needed view to create a SubDataFrame. But I will describe why we have used it and what it is.

Reply:
yes - you might then add a reference that the details about views are discussed later in "such and such" section.